pkg/registry/apis/provisioning/register.go

```diff
@@ -581,6 +616,11 @@ func (b *APIBuilder) Validate(ctx context.Context, a admission.Attributes, o adm
 	if ok {
 		return nil
 	}
+	// FIXME: Do nothing for Connection for now
```

Comment:
I'd say let's add a basic validator for the spec fields themselves. Let's put in `apps/provisioning/pkg/connections`. 

We won't check with Github now, but at least basic that the fields aren't empty, valid type, and other simple things.